### DISCUSSION TITLE
What is the purpose of `--unity=subprojects`?

### DISCUSSION THREAD

**Discussion by bruchar1** [mesonbuild/meson#11416] [Q&A] [answered]
It seems to enable unity build if there are subprojects in the current project. What is the use case of that?

**Comment by jpakkane** [comment, marked as answer]
Recompilations are slower with Unity builds. The assumption here is that you rarely edit subprojects (them being external dependencies) but the top project is edited often. Thus you want to optimize subprojects to build fast from scratch but the top project to build fast incrementally. Which is what you get with that option.
